cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Util.hs

```diff
@@ -258,15 +258,23 @@ cardanoCliPath = "cardano-cli"
 -- | Return the input file path after annotating it relative to the project root directory
 noteInputFile :: (MonadTest m, HasCallStack) => FilePath -> m FilePath
 noteInputFile filePath = GHC.withFrozenCallStack $ do
-  H.annotate $ cardanoCliPath <> "/" <> filePath
+  if isAbsolute filePath
+    then H.annotate filePath
+    else H.annotate $ cardanoCliPath </> filePath
   return filePath
 
 -- | Return the test file path after annotating it relative to the project root directory
 noteTempFile :: (MonadTest m, HasCallStack) => FilePath -> FilePath -> m FilePath
 noteTempFile tempDir filePath = GHC.withFrozenCallStack $ do
-  let relPath = tempDir <> "/" <> filePath
-  H.annotate $ cardanoCliPath <> "/" <> relPath
-  return relPath
+  if isAbsolute filePath
+    then H.note filePath
+    else do
+      let tempWithFilePath = tempDir </> filePath
+      if isAbsolute tempWithFilePath
+        then H.note tempWithFilePath
+        else do
+          H.annotate $ cardanoCliPath </> tempWithFilePath
+          return tempWithFilePath
```

Comment:
Why are we annotating absolute paths, but returning a relative one only in this place? What issue does this function solve?